### PR TITLE
Use property based testing with FsCheck

### PR DESCRIPTION
**Pull request by jpreese**
#251 

I'm honestly not sure what else to add to the HINTS.md file, other than whats already there. You speak to value vs property based already and link to your blog post going in depth about how to use it in practice.

### CODE REVIEW

File: exercises/diamond/Diamond.cs [hunk 1]
```diff
@@ -1,9 +1,12 @@
 ﻿using System;
 
-public static class Diamond
+namespace Diamond
```
Comment (ErikSchierboom) [inline exercises/diamond/Diamond.cs]: Could you remove the namespace? We don't use namespaces in any of the exercises to make things as simple and clean as possible.

File: exercises/diamond/DiamondTest.cs [hunk 2]
```diff
@@ -1,110 +1,131 @@
...
-using System.Collections.Generic;
+﻿using Xunit;
+using FsCheck.Xunit;
+using FsCheck;
+using System;
 using System.Linq;
-using Xunit;
 
-public class DiamondTest
+namespace Diamond
```
Comment (ErikSchierboom) [inline exercises/diamond/DiamondTest.cs]: Could you remove the namespace? We don't use namespaces in any of the exercises to make things as simple and clean as possible.

File: exercises/diamond/Example.cs [hunk 3]
```diff
@@ -1,29 +1,32 @@
 ﻿using System;
 using System.Linq;
 
-public static class Diamond
+namespace Diamond
```
Comment (ErikSchierboom) [inline exercises/diamond/Example.cs]: Could you remove the namespace? We don't use namespaces in any of the exercises to make things as simple and clean as possible.

**Comment by ErikSchierboom** [comment]
@jpreese I've requested some small changes, but great work!

**Comment by jpreese** [comment]
Whoops. I've definitely added namespaces to my previous exercise implementations. I'll go back and submit a new PR to clean those up soon.

**Comment by ErikSchierboom** [comment]
Thanks! 🎉

**Pull request merged**